core/Types.h

```diff
@@ -502,68 +501,105 @@ TYPE_INLINED(LiteralType) final {
     std::string showValue(const GlobalState &gs) const;
     uint32_t hash(const GlobalState &gs) const;
 
-    bool equals(const LiteralType &rhs) const;
+    bool equals(const NamedLiteralType &rhs) const;
     void _sanityCheck(const GlobalState &gs) const;
 };
-CheckSize(LiteralType, 16, 8);
+CheckSize(NamedLiteralType, 8, 8);
```

Comment:
The size of `NamedLiteralType` is only `8` because of the `__attribute__((aligned(8)))` in the class declaration, IIUC.

Do you think it's possible for us to static assert that the size of a class created via `TYPE_INLINED` *without* that attribute is size 4 and thus is small enough to fit into a `TypePtr`?

e.g., right now i don't know if there's anything protecting someone from accidentally adding another `uint32_t` to a `TYPE_INLINED` class, having the `CheckSize` assertion still say "yep, it's 8 bytes" but then having some data get dropped in our bit shifts earlier.

If this is already checked somewhere else please lmk

Reply:
I think this is a good point, but we already have the same sort of issue with `ClassType`, `SelfTypeParam`, and `AliasType`.  `BlamedUntyped` even has the issue that it's eight bytes of data but still can be inlined!

And I think `BlamedUntyped` points at the issue with adding more stuff: eventually you are going to have to construct an instance of the enclosing type and figure out how you're going to get bits out of the type...and how you're going to put bits into the type.  So I think there are already some guardrails against people adding more stuff.